flint/flint-spark-integration/src/main/scala/org/opensearch/flint/spark/FlintSpark.scala

```diff
@@ -144,6 +148,9 @@ class FlintSpark(val spark: SparkSession) {
     }
   }
 
+  private def isIncrementalRefreshing(indexName: String): Boolean =
+    spark.streams.active.exists(_.name == indexName)
```

Comment:
The precondition is streaming index job and full index job are runningin same cluster, then we can detect the illegal state. Add to limitation doc?

Reply:
Fixed in https://github.com/opensearch-project/sql/pull/1800/commits/50610a2ca0b29e7ec88d0452f313ffedc0bd4108. Thanks!